### DISCUSSION TITLE
Tail Log in Terminal

### DISCUSSION THREAD

**Discussion by oaustegard** [espanso/espanso#1261] [Feature requests and Ideas]
**Is your feature request related to a problem? Please describe.**
As an Espanso power user I would like a more efficient way of debugging matches and replacement yaml and script by having a live view of the log file in a terminal window (or equivalent).

**Describe the solution you'd like**
I'd like a `tail` parameter be added to the `espanso log` command with an effect similar to issuing a `tail -f espanso.log` command (from the espanso runtime folder), or `Get-Content -Wait -Path $env:LOCALAPPDATA\espanso\espanso.log` in PowerShell

**Describe alternatives you've considered**
Alternatively perhaps just a baked in way to open a terminal window, execute a statement and leave the terminal window open - then this could just be a baked in espanso match/expansion. My current workaround is a custom match for the PowerShell line above, but it has to be executed manually 😱 

**Additional context**
I see from the implementation of log that it is pretty simplistic - merely reading the file and printing each line - a tail implementation would likely require a from scratch implementation with considerations for the various supported OSes

**Comment by federico-terzi** [comment]
Hey @oaustegard,

Thank you for the feedback! I'll mark it as a feature request :)

Cheers :)

**Comment by oaustegard** [comment]
Just a thought: plausibly rather than the terminal this could be loaded in a window similar to the #acfg# - triggered Active configuration window

**Comment by dougpagani** [comment]
hey federico, in conjunction with tailing, can you give a "verbose mode" for logs, and make it configurable via the yaml? Mostly I want to be able to log to print-out expansions made. Nothing more, really. 
e.g.
> `[INFO] Expansion: ;espanso –> Hi there!` 

Without that data, I won't be able to write a "edit last expanded snippet" script and attach it to a keybinding. I usually have to go hunting any expansion, confused as to where it came from (especially if it is regex). I already have the edit script completely written, I just need a trigger name with which to search the yaml's by.

**Comment by federico-terzi** [comment]
Hey @dougpagani,

Thank you for the feedback! That sounds like an interesting use case, what shortcut would you use?

> in conjunction with tailing, can you give a "verbose mode" for logs, and make it configurable via the yaml? Mostly I want to be able to log to print-out expansions made

I'd love to, but a feature like that would potentially become a security issue, and I really care about keeping Espanso as secure and privacy-oriented as possible. If I added a flag to log the matches, that would make Espanso a keylogger (a rather limited one though), and that's something very dangerous if used in the wrong ways.
So no logging option will probably be added in the foreseeable future. That said, we can definitely evaluate a way to examine the last used match. Please open a separate feature request with some details on the use-case if you're interested :)

Hope you will understand :)

**Comment by smeech** [comment]
#2081 related.
